rest/client/methods.go

```diff
@@ -313,8 +313,35 @@ func (c *communicatorImpl) SendLogMessages(ctx context.Context, taskData TaskDat
 	}
 	info.setTaskPathSuffix("log")
 	var cancel context.CancelFunc
-	ctx, cancel = context.WithTimeout(ctx, 10*time.Minute)
+	ctx, cancel = context.WithDeadline(ctx, time.Now().Add(10*time.Minute))
 	defer cancel()
+	backupTimer := time.NewTimer(15 * time.Minute)
+	start := time.Now()
+	defer backupTimer.Stop()
+	doneChan := make(chan bool)
+	defer func() {
+		doneChan <- true
+		close(doneChan)
+	}()
+	go func() {
```

Comment:
panic handler.

Reply:
is this really neccessary? can we just see if the context was cancled for one reason or another by looking at the error value?

Reply:
Added a panic handler. The thing is I'm adding this because it seems like the context is not getting canceled after 10 mins (or something else strange with contexts) so I'd like to not rely on that.

Reply:
@tychoish It seems pretty clear the context is not getting cancelled, but it should be. John, Eric, and I all agree the code is correct. A 10-minute context is never getting cancelled (thousands of minutes go by), and the timer is not firing. This means either there's something super wrong with the clock or there's a bug in Go 1.8 on OS X.